packages/editor/src/hooks/align.js

```diff
@@ -77,7 +77,17 @@ export const withToolbarControls = createHigherOrderComponent( ( BlockEdit ) =>
 	return ( props ) => {
 		const validAlignments = getBlockValidAlignments( props.name );
 
-		const updateAlignment = ( nextAlign ) => props.setAttributes( { align: nextAlign } );
+		const updateAlignment = ( nextAlign ) => {
+			if ( ! nextAlign ) {
+				const blockType = getBlockType( props.name );
+				const blockDefaultAlign = get( blockType, [ 'attributes', 'align', 'default' ] );
+				if ( blockDefaultAlign ) {
+					props.setAttributes( { align: 'none' } );
```

Comment:
Hi @ZebulanStanphill, when applying the classes we are not setting any class for this alignment and no alignnone should be set.
We can not set the align to undefined if a default alignment exists because default attributes are not saved. So if we set the alignment to undefined (as happened before this PR) for both the default alignment and no alignment the attribute was unset and it would not be possible to differentiate both states.
So in order to differentiate no alignment from default alignment, we add 'none' when alignment is not set. But nothing in the markup changes.

Reply:
Ah, okay. :+1:

Reply:
Can you set `null` instead?

Reply:
I tried but unfortunately, null values are also not persisted in the attribute comments making them behave exactly like undefined.